apps/dashboard/src/main/java/com/akto/action/ApiCollectionsAction.java

```diff
@@ -260,6 +266,27 @@ public String createCollection() {
         this.apiCollections = new ArrayList<>();
         this.apiCollections.add(apiCollection);
 
+        try {
+            int userId = Context.userId.get();
+            int accountId = Context.accountId.get();
+    
+            /*
+             * Since admin has all access, we don't update any collections for them.
+             */
+            RBACDao.instance.getMCollection().updateOne(
+                    Filters.and(
+                            Filters.eq(RBAC.USER_ID, userId),
+                            Filters.eq(RBAC.ACCOUNT_ID, accountId),
+                            Filters.ne(RBAC.ROLE, RBAC.Role.ADMIN.getName())
+                    ),
+                    Updates.addToSet(RBAC.API_COLLECTIONS_ID, apiCollection.getId()),
+                    new UpdateOptions().upsert(false)
+            );
+    
+            UsersCollectionsList.deleteCollectionIdsFromCache(userId, accountId);
```

Comment:
can we add this in delete collections as well, for maintaining atomicity

Reply:
cool